includes/class-wc-payments-woopay-direct-checkout.php

```diff
@@ -71,6 +74,36 @@ public function scripts() {
 			return;
 		}
 
+		// Enqueue the WCPay common config script only if it hasn't been enqueued yet.
+		// This may happen when Direct Checkout is being enqueued on pages that are not the cart page,
+		// such as the home and shop pages.
+		if ( function_exists( 'did_filter' ) && did_filter( 'wcpay_payment_fields_js_config' ) === 0 ) {
+			try {
+				// is_test() throws if the class 'Mode' has not been initialized.
+				$is_test_mode = WC_Payments::mode()->is_test();
+			} catch ( Exception $e ) {
+				// Default to false if the class 'Mode' has not been initialized.
+				$is_test_mode = false;
+			}
+
+			wp_register_script( 'WCPAY_WOOPAY_COMMON_CONFIG', '', [], WCPAY_VERSION_NUMBER, false );
+			wp_localize_script(
+				'WCPAY_WOOPAY_COMMON_CONFIG',
+				'wcpayConfig',
+				[
+					'woopayHost'                    => WooPay_Utilities::get_woopay_url(),
+					'testMode'                      => $is_test_mode,
+					'wcAjaxUrl'                     => WC_AJAX::get_endpoint( '%%endpoint%%' ),
+					'woopaySessionNonce'            => wp_create_nonce( 'woopay_session_nonce' ),
+					'isWooPayDirectCheckoutEnabled' => WC_Payments_Features::is_woopay_direct_checkout_enabled(),
+					'platformTrackerNonce'          => wp_create_nonce( 'platform_tracks_nonce' ),
+					'ajaxUrl'                       => admin_url( 'admin-ajax.php' ),
+					'woopayMinimumSessionData'      => WooPay_Session::get_woopay_minimum_session_data(),
+				]
+			);
+			wp_enqueue_script( 'WCPAY_WOOPAY_COMMON_CONFIG' );
```

Comment:
Since this section is a copy of existing code, can we extract it into a static method and use it in both places? This would help avoid duplication in the codebase.

Reply:
Good call! I applied this change in [8be13dc](https://github.com/Automattic/woocommerce-payments/pull/8903/commits/8be13dccf081603c83c6acc1a4f78140758fc46f).